hq_superset/views.py

```diff
@@ -114,17 +115,16 @@ def refresh_hq_datasource(domain, datasource_id, display_name):
     schema = get_schema_name_for_domain(domain)
     csv_table = Table(table=datasource_id, schema=schema)
     datasource_defn = get_datasource_defn(provider, token, domain, datasource_id)
-    column_dtypes = get_column_dtypes(datasource_defn)
+    column_dtypes, date_columns = get_column_dtypes(datasource_defn)
     try:
         with get_csv_file(provider, token, domain, datasource_id) as csv_file:
             df = pd.concat(
                 pd.read_csv(
                     chunksize=1000,
                     filepath_or_buffer=csv_file,
                     encoding="utf-8",
-                    # Todo; make date parsing work
-                    parse_dates=True,
-                    infer_datetime_format=True,
+                    parse_dates=date_columns,
+                    date_parser=parse_date,
```

Comment:
So pandas will send the `date_columns` through the `parse_date` function when it's trying to read the data into the Dataframe?

Reply:
Exactly. Pandas comes with a [`pd.to_datetime()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.to_datetime.html) function you can use, but it's big and caters for a lot of different formats. We know that all the dates and datetimes from HQ are going to be ISO-8601 format, so we can just use Python's `fromisoformat()`.